feature_engine/selection/eliminate_feature_by_feature_importance.py

```diff
@@ -0,0 +1,234 @@
+
+    Attributes
+    ----------
+
+
+    Methods
+    -------
+
+    fit: finds important features
+
+    transform: removes non-important / non-selected features
+
+    fit_transform: finds and removes non-important features
+
+    """
+
+    def __init__(
+        self,
+        estimator=RandomForestClassifier(),
+        scoring="roc_auc",
+        cv=3,
+        threshold=0.01,
+        variables=None,
+        regression=False
+    ):
+
+        if not isinstance(cv, int) or cv < 1:
+            raise ValueError("cv can only take positive integers bigger than 1")
+
+        if not isinstance(threshold, (int, float)):
+            raise ValueError("threshold can only be integer or float")
+
+        if not isinstance(regression, bool):
+            raise ValueError("regression should be a boolean")
+
+        self.variables = _define_variables(variables)
+        self.estimator = estimator
+        self.scoring = scoring
+        self.threshold = threshold
+        self.cv = cv
+        self.regression = regression
+
+    def fit(self, X, y):
+        """
+
+        Args
+        ----
+
+        X: pandas dataframe of shape = [n_samples, n_features]
+           The input dataframe
+
+        y: array-like of shape (n_samples)
+           Target variable. Required to train the estimator.
+
+
+        Returns
+        -------
+
+        self
+        """
+
+        # check input dataframe
+        X = _is_dataframe(X)
+
+        # reset the index
+        X = X.reset_index(drop=True)
+        y = y.reset_index(drop=True)
+
+        # find numerical variables or check variables entered by user
+        self.variables = _find_numerical_variables(X, self.variables)
+
+        # train model with all features and cross-validation
+        model = cross_validate(
+            self.estimator,
+            X,
+            y,
+            cv=self.cv,
+            scoring=self.scoring,
+        )
+
+        # store initial model performance
+        self.initial_model_performance_ = model["test_score"].mean()
+
+        # get performance metric
+        scorer = get_scorer(self.scoring)
+
+        # choose the approriate attibute to use for the appropriate model
+        if (self.regression):
+            get_feature_importance_method = "coef_"
+        else:
+            get_feature_importance_method = "feature_importances_"
+
+        feature_importance_score = {}
+
+        for m in model["estimator"]:
+
+            features = pd.DataFrame(data=getattr(
+                m, get_feature_importance_method), columns=["coef_importance"])
+            features.index = self.variables
+            features.sort_values(ascending=True, inplace=True, by='coef_importance')
+
+            features['variable_score'] = np.arange(features.shape[0])
+
+            tmp = features['variable_score'].to_dict()
+            feature_importance_score = {k: tmp.get(
+                k, 0) + feature_importance_score.get(k, 0) for k in set(tmp)}
```

Comment:
I like this approach better, thanks!

Since i have created in my first version an attribute that returns the ordered features by importance list, called "feature_importances_", do you we should get rid of it if we are also returning the whole ordered series with features and feature importance values?

It might be redundant if we keep both. What do you think ?

Reply:
If it has the same information, we should also keep one of the 2.